actors/evm/src/interpreter/precompiles.rs

```diff
@@ -179,28 +382,140 @@ fn lookup_address<RT: Runtime>(rt: &RT, input: &[u8]) -> PrecompileResult {
 /// Reads a FIL encoded address
 /// Resolves a FIL encoded address into an ID address
 /// returns BE encoded u64 or empty array if nothing found
-fn resolve_address<RT: Runtime>(rt: &RT, input: &[u8]) -> PrecompileResult {
-    let addr = match Address::from_bytes(input) {
+fn resolve_address<RT: Runtime>(rt: &RT, input: &[u8], _: PrecompileContext) -> PrecompileResult {
+    let mut input_params = U256Reader::new(input);
+
+    let len = input_params.next_param_padded::<u32>()? as usize;
+    let addr = match Address::from_bytes(&read_right_pad(input_params.remaining_slice(), len)) {
```

Comment:
It looks like reading then padding is common. We might want to provide a method.

Reply:
later if we need it, this PR already has a lot of changes